internal/request/method_with_insert.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Searis AS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package request
+
+import (
+	"context"
+
+	"github.com/clarify/clarify-go/jsonrpc"
+)
+
+const (
+	includeParam jsonrpc.ParamName = "include"
+)
+
+// Method is a constructor for Requests against a given RPC method.
+type IncludeMethod[R any] struct {
+	APIVersion string
+	Method     string
+}
+
+func (cfg IncludeMethod[R]) NewRequest(h jsonrpc.Handler, params ...jsonrpc.Param) WithInclude[R] {
+	return WithInclude[R]{
+		parent: Request[R]{
+			apiVersion: cfg.APIVersion,
+			method:     cfg.Method,
+
+			baseParams: params,
+			h:          h,
+		},
+	}
+}
+
+// WithInclude describe an initialized RPC request with access to a request
+// handler and the option to include related resources.
+type WithInclude[R any] struct {
```

Comment:
`WithXxx` is expected to be a method. Maybe we can try `IncludeRel` instead?

Reply:
This is an internal type, so the name should not be visible in the external interface.

Reply:
I'll rename it.

Reply:
`request.Relational` maybe?

Reply:
Tired `request.Relational` / `request.RelationalMethod`. Notice that the file has been renamed. Let me know what you think.